explorer/explorer.go

```diff
@@ -172,7 +179,19 @@ func NewExplorer(cm ChainManager, store Store, indexCfg config.Index, scanCfg co
 				e.log.Error("failed to get tip", zap.Error(err))
 			}
 			if err := e.syncStore(lastTip, indexCfg.BatchSize); err != nil {
-				e.log.Panic("failed to sync store", zap.Error(err))
+				switch {
+				case errors.Is(err, context.Canceled):
+					return
+				case strings.Contains(err.Error(), "missing block at index"):
+					log.Warn("missing block at index, resetting chain state", zap.Stringer("id", lastTip.ID), zap.Uint64("height", lastTip.Height))
+					if err := e.s.ResetChainState(); err != nil {
+						log.Panic("failed to reset explorer state", zap.Error(err))
+					}
+					// trigger resync
+					reorgChan <- types.ChainIndex{}
```

Comment:
this potentially deadlocks the loop I believe, should do select with a default case to ensure it doesn't block on this